include/gz/sim/MeshInertiaCalculator.hh

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (C) 2023 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef GZ_SIM_MESHINERTIACALCULATOR_HH_
+#define GZ_SIM_MESHINERTIACALCULATOR_HH_
+
+#include <optional>
+#include <vector>
+
+#include <sdf/sdf.hh>
+#include <sdf/CustomInertiaCalcProperties.hh>
+
+#include <gz/sim/Util.hh>
+#include <gz/sim/Export.hh>
+
+#include <gz/common/graphics.hh>
+#include <gz/common/Mesh.hh>
+#include <gz/common/MeshManager.hh>
+
+#include <gz/math/Vector3.hh>
+#include <gz/math/MassMatrix3.hh>
+#include <gz/math/Inertial.hh>
+
+
+namespace gz
+{
+  namespace sim
+  {
+    // Inline bracket to help doxygen filtering.
+    inline namespace GZ_SIM_VERSION_NAMESPACE
+    {
+      /// \struct Triangle gz/sim/MeshInertiaCalculator.hh
+      /// \brief A struct to represent a triangle of the mesh
+      /// An instance of the struct holds 3 vector3d instances
+      /// each of which represents a vertex of the triangle
+      struct Triangle
+      {
+        gz::math::Vector3d v0, v1, v2;
+      };
+
+      /// \class MeshInertiaCalculator gz/sim/MeshInertiaCalculator.hh
+      /// \brief Inertial Properties (Mass, Centre of Mass & Moments of
+      /// Inertia) calculator for 3D meshes.
+      ///
+      /// This class overloads the () operator, therefore, an instance
+      /// of this class registered with libsdformat as a Custom
+      /// Inertia Calculator. This would used to calculate the
```

Comment:
```suggestion
      /// Inertia Calculator. This would use to calculate the
```

Reply:
Corrected in [a2a1cba](https://github.com/gazebosim/gz-sim/pull/2061/commits/a2a1cba22551449b3afa2c276d0e6bd02271f780)